sentinelhub/api/batch_statistical.py

```diff
@@ -0,0 +1,148 @@
+"""
+Module implementing an interface with
+`Sentinel Hub Batch Processing API <https://docs.sentinel-hub.com/api/latest/api/batch-statistical/>`__.
+"""
+import datetime as dt
+import logging
+from dataclasses import dataclass, field
+from typing import Any, Optional, Union
+
+from dataclasses_json import CatchAll, LetterCase, Undefined
+from dataclasses_json import config as dataclass_config
+from dataclasses_json import dataclass_json
+
+from ..type_utils import Json, JsonDict
+from .batch_base import BaseBatchClient, BaseBatchRequest, BatchRequestStatus, BatchUserAction
+from .utils import datetime_config, enum_config, remove_undefined
+
+LOGGER = logging.getLogger(__name__)
+
+BatchStatisticalRequestType = Union[str, dict, "BatchStatisticalRequest"]
+
+
+class SentinelHubBatchStatistical(BaseBatchClient["BatchStatisticalRequest"]):
+    """An interface class for Sentinel Hub Batch Statistical API
+
+    Check `Batch Statistical API <https://docs.sentinel-hub.com/api/latest/reference/#tag/batch_statistical>`__ for more
+    information.
+    """
+
+    # pylint: disable=too-many-public-methods
+    @staticmethod
+    def _get_service_url(base_url: str) -> str:
+        """Provides URL to Batch Statistical API"""
+        return f"{base_url}/api/v1/statistics/batch"
+
+    def create(
+        self,
+        input_data: JsonDict,
+        input_features: JsonDict,
+        aggregation: JsonDict,
+        calculations: JsonDict,
+        output: JsonDict,
+        **kwargs: Any,
+    ) -> "BatchStatisticalRequest":
+        """Create a new batch statistical request
+
+        `Batch Statistical API reference
+        <https://docs.sentinel-hub.com/api/latest/reference/#operation/createNewBatchStatisticsRequest>`__
+        """
+
+        payload = {
+            "input": {"features": input_features, "data": input_data},
+            "aggregation": aggregation,
+            "calculations": calculations,
+            "output": output,
+            **kwargs,
+        }
+        payload = remove_undefined(payload)
+
+        url = self.service_url
+        request_info = self.client.get_json(url, post_values=payload, use_session=True)
+
+        return BatchStatisticalRequest.from_dict(request_info)  # type: ignore[attr-defined]
```

Comment:
Do you think we could add just a signature for methods `from_dict` and `to_dict` to all classes that are wrapped with `@dataclass_json`? This way the typing and autocomplete would start working.

Reply:
done